src/EFCore.PG/Extensions/NpgsqlPropertyBuilderExtensions.cs

```diff
@@ -611,5 +612,133 @@ public static IConventionPropertyBuilder UseIdentityByDefaultColumn([NotNull] th
+        /// <param name="config">
+        /// <para>
+        /// The text search configuration for this generated tsvector property, or <c>null</c> if this is not a
+        /// generated tsvector property.
+        /// </para>
+        /// <para>
+        /// See https://www.postgresql.org/docs/current/textsearch-controls.html for more information.
+        /// </para>
+        /// </param>
+        /// <param name="includedPropertyNames">An array of property names to be included in the tsvector.</param>
+        /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+        /// <returns>
+        /// The same builder instance if the configuration was applied,
+        /// <c>null</c> otherwise.
+        /// </returns>
+        public static IConventionPropertyBuilder IsGeneratedTsVector(
+            [NotNull] this IConventionPropertyBuilder propertyBuilder,
+            [NotNull] string config,
+            [NotNull] IReadOnlyList<string> includedPropertyNames,
+            bool fromDataAnnotation = false)
+        {
+            if (propertyBuilder.CanSetIsGeneratedTsVector(config, includedPropertyNames, fromDataAnnotation))
+            {
+                propertyBuilder.HasColumnType("tsvector");
+                propertyBuilder.Metadata.SetGeneratedTsVectorConfig(config, fromDataAnnotation);
+                propertyBuilder.Metadata.SetGeneratedTsVectorProperties(includedPropertyNames, fromDataAnnotation);
+
+                return propertyBuilder;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Returns a value indicating whether the property can be configured as a full-text search tsvector column.
+        /// </summary>
+        /// <param name="propertyBuilder">The builder for the property being configured.</param>
+        /// <param name="config">
+        /// <para>
+        /// The text search configuration for this generated tsvector property, or <c>null</c> if this is not a
+        /// generated tsvector property.
+        /// </para>
+        /// <para>
+        /// See https://www.postgresql.org/docs/current/textsearch-controls.html for more information.
+        /// </para>
+        /// </param>
+        /// <param name="includedPropertyNames">An array of property names to be included in the tsvector.</param>
+        /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+        /// <returns><c>true</c> if the property can be configured as a full-text search tsvector column.</returns>
+        public static bool CanSetIsGeneratedTsVector(
+            [NotNull] this IConventionPropertyBuilder propertyBuilder,
+            [NotNull] string config,
+            [CanBeNull] IReadOnlyList<string> includedPropertyNames,
+            bool fromDataAnnotation = false)
+        {
+            Check.NotNull(propertyBuilder, nameof(propertyBuilder));
```

Comment:
Why do you check `propertyBuilder` only here?

Reply:
Made this more coherent, thanks.